src/components/carousel/nz-carousel.component.ts

```diff
@@ -83,14 +96,18 @@ export class NzCarouselComponent implements AfterViewInit, OnDestroy {
       if (this.nzAutoPlay) {
         this.createInterval();
       }
-      this._renderer.setStyle(this.slickList.nativeElement, 'height', `${this.hostElement.nativeElement.offsetHeight}px`);
+
       if (this.nzVertical) {
+        this._renderer.setStyle(this.slickList.nativeElement, 'height', `${this.hostElement.nativeElement.offsetHeight}px`);
```

Comment:
One line of code is missing here.
`this._renderer.removeStyle(this.slickList.nativeElement, 'height');`

if not，in vertical mode, the slickList's height will be setting to  0 .